pkg/provider/azure_storageaccount.go

```diff
@@ -160,6 +170,17 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 		subsID = accountOptions.SubscriptionID
 	}
 
+	if len(accountOptions.StorageEndpointSuffix) == 0 {
+		accountOptions.StorageEndpointSuffix = az.cloud.Environment.StorageEndpointSuffix
+	}
+
+	if accountOptions.CreatePrivateEndpoint {
+		if accountOptions.StorageType == "" {
```

Comment:
to make it backward compatible, if `StorageType == ""`, it should be `file` by default.